Public/Src/Engine/Scheduler/Graph/PipGraph.Builder.cs

```diff
@@ -57,6 +58,10 @@ public class Builder : PipGraphBase, IPipGraphBuilder
 
             private readonly IConfiguration m_configuration;
 
+            private WindowsOsDefaults m_windowsOsDefaults;
+            private MacOsDefaults m_macOsDefaults;

```

Comment:
nit: you could make these `Lazy<WindowsOsDefaults>` and `Lazy<MacOsDefaults>` to avoid explicitly locking and double checking  in `ApplyCurrentOsDefaults`

Reply:
I can't because I need a pathtable to initialize them properly...

---
In reply to: [274228802](https://github.com/Microsoft/BuildXL/pull/125#discussion_r274228802) [](ancestors = 274228802)